human-dynamics-estimator/src/HumanDynamicsEstimator.cpp

```diff
@@ -68,53 +176,293 @@ bool HumanDynamicsEstimator::configure(yarp::os::ResourceFinder &rf)
+            BerdyOutputRangeMap &map = m_link_map[berdyDyn[index].id];
+            map[tempTYPE] = berdyDyn[index].range;
+        }
+        if (berdyDyn[index].type==iDynTree::JOINT_WRENCH) {
+            tempTYPE = "transmittedWrench";
+            BerdyOutputRangeMap &map = m_link_map[berdyDyn[index].id];
+            map[tempTYPE] = berdyDyn[index].range;
+        }
+        if (berdyDyn[index].type==iDynTree::NET_EXT_WRENCH) {
+            tempTYPE = "externalWrench";
+            BerdyOutputRangeMap &map = m_link_map[berdyDyn[index].id];
+            map[tempTYPE] = berdyDyn[index].range;
+        }
+        if (berdyDyn[index].type==iDynTree::DOF_TORQUE) {
+            tempTYPE = "torque";
+            BerdyOutputRangeMap &map = m_joint_map[berdyDyn[index].id];
+            map[tempTYPE] = berdyDyn[index].range;
+        }
+        if (berdyDyn[index].type==iDynTree::DOF_ACCELERATION) {
+            tempTYPE = "acceleration";
+            BerdyOutputRangeMap &map = m_joint_map[berdyDyn[index].id];
+            map[tempTYPE] = berdyDyn[index].range;
+        }
+    }    
+
+    /* 
+     * ------Open and prepare a port:o for the output
+     */     
+    if (! m_outputPort.open("/humanDynamics:o"))
+    {
+        yError() << "Unable to open port /humanDynamics:o";
+        return false;
+    } 
     
     return true;
 }
 
 //---------------------------------------------------------------------
 bool HumanDynamicsEstimator::updateModule()
-{
-    count++;
-    std::cout<<"["<<count<<"]"<< " updateModule:< HumanDynamicsEstimator >"<<std::endl;
-    return true;
+{   
+    m_count ++;
     
-    // TODO: MAP computation --> Maria's code to be integrated
-}
+    /*
+    * ------Read from ports
+    */
+    human::HumanState *tmp_state = m_humanJointConfiguration_port.read();
+            
+    if (tmp_state == NULL)
+    {
+        yError("Something wrong in reading the human configuration port!");
+        return false;
+    }
+            
+    // transform yarp Vector in iDynTree vector
+    if (!iDynTree::toiDynTree(tmp_state->positions,m_q))
+    {
+        yError("Somenthing wrong in the conversion from a YARP to iDynTree vector!");
+        return false;
+    };
 
-//---------------------------------------------------------------------
-bool HumanDynamicsEstimator::interruptModule()
-{
-    return true;
-}
+    // transform yarp Vector in iDynTree vector           
+    if (!iDynTree::toiDynTree(tmp_state->velocities,m_dq))
+    {
+         yError("Somenthing wrong in the conversion from a YARP to iDynTree vector!");
+         return false;
+    };
 
+    human::HumanForces *tmp_force = m_humanForces_port.read();
+
+    if (tmp_force == NULL)
+    {
+        yError("Something wrong in reading the human forces port!");
+        return false;
+    }
+    
+    for (std::vector<human::Force6D>::iterator it(tmp_force->forces.begin()); it!=tmp_force->forces.end(); ++it ) 
+    {
+        std::string appliedLink = it->appliedLink;
```

Comment:
If you declare a reference (or const reference to be safe) you avoid a copy of a string